web_tree_duplicate/README.rst

```diff
@@ -0,0 +1,116 @@
+===========================
+Tree View Duplicate Records
+===========================
+
+.. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fweb-lightgray.png?logo=github
+    :target: https://github.com/OCA/web/tree/12.0/web_tree_duplicate
+    :alt: OCA/web
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/web-12-0/web-12-0-web_tree_duplicate
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/162/12.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+With this module you can duplicate records directly from the tree view.
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+The duplicate option is disabled by default.
+To enable it you have to add attribute `duplicate` to the tree view.
```

Comment:
Shouldn't this be called `multi_duplicate` for avoiding misunderstandings?

Reply:
Hmm yes, that's better. 👍  I'll also try to make it filter on the new copied records.

Reply:
Tested it. multi_duplicate gives an error:
`Invalid view account.analytic.line.tree.hr_timesheet definition in False`
The attribute `duplicate` is also used in the form view, so I'm keeping it `duplicate` it's more consistent and no extra python code is needed.

Reply:
That's because you have to change the RNG that indicates the allowed structure:

https://github.com/odoo/odoo/blob/c9d3653f486f6c22b47c91f9dc7583219fffc47b/odoo/addons/base/rng/tree_view.rng#L17

I'm not sure if we can inherit that file. But if we reuse the existing attribute, we are overwriting other possible functionality and this might fail in the future.

Reply:
It's possible to inherit :( But I checked the web.ListView, and web.ListRenderer and it doesn't use the duplicate attribute. For now it's ok to use it IMO. In another version of Odoo this module probably gets deprecated.

Reply:
Is it or is not possible? (your sad face confused me), but if possible, I prefer to do it at once than to be exposed in later versions or in this one with an overlap.

Reply:
not possible :smile:

Reply:
OK, that's something limiting indeed, so another option is to use `options` attribute. I let you in your hands the path to take for not being more picky.